p2panda-rs/src/operation/operation_meta.rs

```diff
@@ -88,6 +90,20 @@ impl AsOperation for OperationWithMeta {
     }
 }
 
+impl CoreHash for OperationWithMeta {
+    fn hash<H: Hasher>(&self, state: &mut H) {
+        self.operation_id().as_str().hash(state);
```

Comment:
This is returning the hash of the string and not of the operation (aka the id hash) if I see this right

Reply:
Yes, it is... This is my first time implementing equality traits, and I went for what seemed like the simplest way to establish equality. If the hashes are equal, the operations are equal. Maybe this is a massive abuse of the trait. In which case I need to apologise to the rust god's 😂

Reply:
Oh, but this is the hash trait we're talking about. Kind of same line of thinking though...

Reply:
The docs seem to suggest hashing only some key values in a struct is valid: https://doc.rust-lang.org/std/hash/trait.Hash.html

Reply:
The point is that you already have a hash in `instance_id` and you turn this into a string and hash it *again*. I wonder if it would simply be enough the return what is inside of `instance_id` or move the whole hashing logic of `Operation.hash()` there (https://github.com/p2panda/p2panda/issues/32)

Reply:
Ah, haha, yeh this is silly. I would like to move the hashing logic to `Hash`, this makes sense to me.

Reply:
in progress here: https://github.com/p2panda/p2panda/pull/178